### PR TITLE
Parameter validation: Raises error for all missing

### LINKED ISSUE #802: Report all missing required parameters instead of just the first
In the case of an endpoint having multiple required params, would it be possible to include all missing params in the error message? As it is, it looks like these are reported one-by-one in order:

https://github.com/Apipie/apipie-rails/blob/6a0174d20d3ccfe9a0a528826eb4abe886728021/lib/apipie/dsl_definition.rb#L246-L251

### PR DESCRIPTION
**Pull request by davidwessman**
- Instead of just raising `ParamMissing` for the first missing,
  instead raise a compound `ParamMultipleMissing` if there are more
  than one missing.
- Adds specs for both POST and GET requests.
- Fixes #802

### CODE REVIEW

File: lib/apipie/errors.rb [hunk 1]
```diff
@@ -24,6 +24,20 @@ def initialize(param)
     end
   end
 
+  class ParamMultipleMissing < ParamError
```
Comment (davidwessman) [inline lib/apipie/errors.rb:27]: Would be nice to inherit from the same error as `ParamMissing`, but we do not have the `defined param` here.

**Comment by davidwessman** [comment]
I have no good way to solve the last rubocop warnings, do you usually add ignores if the complexity grows? Not sure if I can split it up more since it is meta-programming

**Comment by PanosCodes** [comment]
Running `rubocop --auto-gen-config --exclude-limit 180` will add the errors to `.rubocop_todo.yml`

**Review by mathieujobin — APPROVED** [review]
Yep, its OK to ignore `ModuleLength` and `BlockLength` rubocop offenses, you can use the comment as @PanosCodes suggested. and I will merge.

Thank you

**Review by mathieujobin — APPROVED** [review]

**Comment by davidwessman** [comment]
Thanks! 🙂

**Pull request merged**